### ISSUE TITLE
Encountered error: gob: encoder: message too big

### ISSUE THREAD

**Issue by boulund** [will-rowe/groot#10]
Hi!

Unfortunately having some trouble running groot. Based on the log file (below), it appears to fail when trying to save the index. I ran:

    groot get --database groot-db --out /db/groot/groot-db/ -y /db/groot/groot-db.log
    groot index -i /db/groot/groot-db/groot-db.90 --readLength 125 --processors 30
    
```
2018/10/12 12:23:18 i am groot (version 0.8.1)                                                    
2018/10/12 12:23:18 starting the index subcommand                                                 
2018/10/12 12:23:18 checking parameters...                                                        
2018/10/12 12:23:18     indexing scheme: lshForest                                                
2018/10/12 12:23:18     processors: 30                                                            
2018/10/12 12:23:18     k-mer size: 7                                                             
2018/10/12 12:23:18     signature size: 128                                                       
2018/10/12 12:23:18     read length (window size): 125                                            
2018/10/12 12:23:18     number of MSA files found: 1156                                           
2018/10/12 12:23:18 building groot graphs...                                                      
2018/10/12 12:23:20     number of groot graphs built: 1156                                        
2018/10/12 12:23:20 windowing graphs and generating MinHash signatures...                         
2018/10/12 12:23:39     number of signatures generated: 3525379                                   
2018/10/12 12:23:39 running LSH Forest...                                                         
2018/10/12 12:23:53     number of hash functions per bucket: 128                                  
2018/10/12 12:23:53     number of buckets: 1                                                      
2018/10/12 12:23:53 saving index files to "./groot-index-20181012122318"...                       
2018/10/12 12:24:07 encountered error: gob: encoder: message too big                              
```

**Comment by will-rowe** [comment]
That will teach me to break the no commit Friday rule!

Im struggling to recreate this though, what groot version and what system are you using?

**Comment by boulund** [comment]
Sorry, I should have put that in the inital post. I'm running `0.8.1` from conda.

**Comment by boulund** [comment]
It works fine to download and build `arg-annot`, though.

**Comment by will-rowe** [comment]
Okay - will dig into this, the gob encoding is Go’s binary format, will see what the message size limit is and how it is determined/set. Glad it works for argannot, means that it is probably just the index from the groot dB is too big. Will get to this on Monday. Have a good weekend.

**Comment by will-rowe** [comment]
Okay - looks like Go's GOB isn't really designed for this, it is more for message passing and has a maximum limit for message size. I could change the limit but this isn't a good fix, someone might want to use groot on even bigger databases.

I will look into alternative serialisation patterns, maybe JSON or binary encoding. Will work on this now!

**Comment by boulund** [comment]
Sorry to hear it's not just a quick fix. I'm sure you'll find something else to use fairly quickly though. 

I typically always put my data in an SQLite3 database if I need to persist stuff on disk. It's feature-filled, performant, available pretty much everywhere, and pretty future-proof. Maybe that would work for you in this case as well?

**Comment by will-rowe** [comment]
Thanks for the suggestion - went for messagepack in the end as I thought SQL was a bit of overkill at the moment, but I do like the sound of adding this as an option in the future.

[Messagepack]() ends up being much more performant than GOB and is on a par with BSON, so this has worked out well! Fix is currently in dev branch (92e28d582f39695be2449cbbbf79d6f922ecc628) but will merge to master once I have fixed the other issue you raised!

Thanks again

**Comment by will-rowe** [comment]
Okay - have merged this to master and released it. Should be in conda shortly.
